cfme/tests/infrastructure/test_providers.py

```diff
@@ -256,15 +253,11 @@ def custom_attributes(self, rest_api, infra_provider):
 
         yield attrs, provider
 
-        log_warn = False
-        for attr in attrs:
-            try:
-                provider.custom_attributes.action.delete(attr)
-            except APIException:
-                # custom attributes can be deleted by tests, just log warning
-                log_warn = True
-        if log_warn:
-            logger.warning("Failed to delete custom attribute.")
+        provider.custom_attributes.reload()
+        ids = [attr.id for attr in attrs]
+        delete_attrs = [attr for attr in provider.custom_attributes if attr.id in ids]
+        if len(delete_attrs) != 0:
```

Comment:
PEP8 "For sequences, (strings, lists, tuples), use the fact that empty sequences are false."

```
if delete_attrs:
    ...
```